src/canvas-view/canvas-editor/toolbox-items.component.js

```diff
@@ -0,0 +1,27 @@
+(function () {
+  'use strict';
+
+  angular.module('patternfly.canvas')
+    .component('toolboxItems', {
+      templateUrl: 'canvas-view/canvas-editor/toolbox-items.html',
+      bindings: {
+        items: '=',
+        startDragCallback: '=',
+        clickCallback: '=',
+        searchText: '='
+      },
+      controller: function toolboxItemsController () {
+        var ctrl = this;
+
+        ctrl.clickCallbackfmDir = function (item) {
+          if (!item.disableInToolbox) {
+            ctrl.clickCallback(item);
```

Comment:
This may have been in the existing directive, but if we're going for consistency...

we do checks for a function within the ```node-toolbar-component.js``` should we be doing that here too?
Similar too...
```js
if (angular.isFunction(ctrl.clickCallback)) {...
```

Reply:
Can do. Makes sense for consistency purposes.

Reply:
After looking into it, seems like ```ctrl.clickCallback ``` isn't a function so would be unable to check for a function in this instance

Reply:
If it isn't a function, how is it being called?

Reply:
Or maybe I'm misunderstanding